generators/entity-server/files.js

```diff
@@ -386,22 +386,36 @@ function writeFiles() {
 
         writeEnumFiles() {
             this.fields.forEach(field => {
-                if (field.fieldIsEnum === true) {
-                    const fieldType = field.fieldType;
-                    const enumInfo = utils.buildEnumInfo(field, this.angularAppName, this.packageName, this.clientRootFolder);
-                    if (!this.skipServer) {
-                        this.template(
-                            `${this.fetchFromInstalledJHipster(
-                                'entity-server/templates'
-                            )}/${SERVER_MAIN_SRC_DIR}package/domain/enumeration/Enum.java.ejs`,
-                            `${SERVER_MAIN_SRC_DIR}${this.packageFolder}/domain/enumeration/${fieldType}.java`,
-                            this,
-                            {},
-                            enumInfo
-                        );
-                    }
+                if (!field.fieldIsEnum) {
+                    return;
+                }
+                const fieldType = field.fieldType;
+                const enumInfo = {
+                    ...utils.buildEnumInfo(field, this.clientRootFolder),
+                    angularAppName: this.angularAppName,
+                    packageName: this.packageName
+                };
+                if (!this.skipServer) {
+                    this.template(
+                        getTemplateFileForEnums(this.fetchFromInstalledJHipster('entity-server/templates'), enumInfo),
+                        `${SERVER_MAIN_SRC_DIR}${this.packageFolder}/domain/enumeration/${fieldType}.java`,
+                        this,
+                        {},
+                        enumInfo
+                    );
                 }
             });
         }
     };
 }
+
+function getTemplateFileForEnums(jhipsterInstallationPath, enumInfo) {
+    const pathToEnumFolder = `${jhipsterInstallationPath}/${SERVER_MAIN_SRC_DIR}package/domain/enumeration`;
+    if (enumInfo.withoutCustomValues) {
```

Comment:
I don’t like the idea of creating 2 templates.
Usually the template is the same:
```
If (withCustomValues ) {
With custom values  template
} else {
Without
}
```

Reply:
The issue is that the differences are quite clear between the templates and merging them is gonna lead to a messy result (yes, I find most templates messy but still... :))

Reply:
I'm gonna push the one-file solution